.github/workflows/ci-actions-incremental.yml

```diff
@@ -129,17 +131,31 @@ jobs:
       - name: Generate cache key
         id: cache-key
         run: |
+          CURRENT_BRANCH="${{ github.repository != 'quarkusio/quarkus' && 'fork' || github.base_ref || github.ref_name }}"
+          CURRENT_MONTH=$(/bin/date -u "+%Y-%m")
+          CURRENT_DAY=$(/bin/date -u "+%d")
+          ROOT_CACHE_KEY="m2-cache"
+          echo "m2-monthly-cache-key=${ROOT_CACHE_KEY}-${CURRENT_MONTH}" >> $GITHUB_OUTPUT
+          echo "m2-monthly-branch-cache-key=${ROOT_CACHE_KEY}-${CURRENT_MONTH}-${CURRENT_BRANCH}" >> $GITHUB_OUTPUT
+          echo "m2-cache-key=${ROOT_CACHE_KEY}-${CURRENT_MONTH}-${CURRENT_BRANCH}-${CURRENT_DAY}" >> $GITHUB_OUTPUT
```

Comment:
Wonder if we can create an action to do that for us

Reply:
Yeah I had the same thought. I might do it at some point. For now, I think it's fine even if a bit repetitive.

Ideally I think we would even need to include the cache call into it.

Reply:
Also Quarkus GitHub Actions are composite actions so we could even bundle the cache call into it.